product-scenarios/10-manage-security-resiliancy-other-aspects-of-service-endpoints/10.1-expose-service-with-secured-backend/10.1.1-expose-service-secured-via-basic-authentication/src/test/java/org/wso2/am/scenario/tests/basic/auth/security/BasicAuthEndpointSecuredAPITestcase.java

```diff
@@ -0,0 +1,181 @@
+        String storeURL = infraProperties.getProperty(STORE_URL);
+        String warFileName = "jaxrs_basic";
+        String warFileLocation = System.getProperty("user.dir") + File.separator+ "src/test/resources" + File.separator
+                + warFileName + ".war";
+        String serviceEndpoint = "https://localhost:9443/services/";
+
+        if (publisherURL == null) {
+            publisherURL = "https://localhost:9443/publisher";
+        }
+
+        if (storeURL == null) {
+            storeURL = "https://localhost:9443/store";
+        }
+
+        setKeyStoreProperties();
+        apiPublisher = new APIPublisherRestClient(publisherURL);
+        apiPublisher.login(admin, admin);
+        apiStore = new APIStoreRestClient(storeURL);
+        apiStore.login(admin, admin);
+        try {
+            WebAppDeployUtils.deployWebApp(serviceEndpoint, admin, admin, warFileLocation,
+                    warFileName);
+            log.info("WebApp deployed successfully");
+        } catch (RemoteException | MalformedURLException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test(description = "10.1.1.1")
+    public void testInvokeAPIWithBasicAuthEndpointSecurity() throws Exception {
+        // Create an API
+        apiRequest = new APIRequest(apiName, apiContext, apiVersion, endpointType, endpointAuthType,
+                APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, endpointURL, epUsername, URLEncoder.encode(epPassword, UTF_8), "0",
+                APIMIntegrationConstants.HTTP_VERB_GET,
+                APIMIntegrationConstants.RESOURCE_AUTH_TYPE_APPLICATION_AND_APPLICATION_USER,
+                APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED, apiResource);
+        HttpResponse apiCreationResponse = apiPublisher.addAPI(apiRequest);
+        verifyResponse(apiCreationResponse);
+
+        // Change the API lifecycle state from CREATED to PUBLISHED
+        APILifeCycleStateRequest apiLifeCycleStateRequest = new APILifeCycleStateRequest(apiName, admin,
+                APILifeCycleState.PUBLISHED);
+        HttpResponse apiPublishResponse = apiPublisher.changeAPILifeCycleStatus(apiLifeCycleStateRequest);
+        verifyResponse(apiPublishResponse);
+        log.info("Successfully published the API - " + apiName);
+
+        // Retrieve the created API
+        HttpResponse getAPIResponse = apiPublisher.getAPI(apiName, admin);
+        verifyResponse(getAPIResponse);
+        JSONObject getAPIRespData = new JSONObject(getAPIResponse.getData());
+        assertTrue(getAPIRespData.getJSONObject("api").get("endpointAuthTypeDigest").equals("false"),
+                "Endpoint security type does not match");
+        assertTrue(getAPIRespData.getJSONObject("api").get("epUsername").equals(epUsername),
+                "Endpoint username does not match");
+        assertTrue(getAPIRespData.getJSONObject("api").get("epPassword").equals(epPassword),
+                "Endpoint password does not match");
+
```

Comment:
After publishing the API, it will take time to appear in store. We have included a method for this in parent class. "isAPIVisibleInStore". Can you use it here?

Reply:
Added "isAPIVisibleInStore" method prior to subscribe the API